### PR TITLE
Log distributed data size, #29736

### PR DESCRIPTION
**Pull request by patriknw**
References #29736

### CODE REVIEW

File: akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import akka.annotation.InternalApi
+import akka.cluster.ddata.Key.KeyId
+import akka.event.LoggingAdapter
+
+/**
+ * INTERNAL API
+ *
+ * This class is not thread-safe. It is supposed to be used from an actor.
+ */
+@InternalApi private[akka] class LogPayloadSize(log: LoggingAdapter, logSizeExceeding: Int, warnSizeExceeding: Int) {
```
Comment (patriknw) [inline akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala:16]: this is similar to the existing LoggingRemoteInstrument

File: akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      // 10% threshold until next log
+      def newMax = (size * 1.1).toInt
+
+      maxPayloadBytes.get(key) match {
+        case Some(max) =>
+          if (size > max) {
+            maxPayloadBytes = maxPayloadBytes.updated(key, newMax)
+            if (size >= warnSizeExceeding)
+              log.warning(
+                "New distributed data size for [{}] is [{}] bytes. Close to max remote message payload size.",
```
Comment (johanandren) [inline akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala:30]: I find the log message a bit misleading, it sounds like something is being reconfigured, what about: "New max distributed data size [{}] bytes seen for [{}]" or something along those lines?

**Review by johanandren — APPROVED** [review]
LGTM with one log message improved.

File: akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala (hunk 2 above)
Comment (octonato) [reply akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala:30]: It's the word 'new' that is causing this confusion? Maybe we should find something.

Comment (patriknw) [reply akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala]: I have updated the log message, and also using same log message for the first and subsequent logs

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**